Migrate license, collections and common modules test to junit 5

MishaDemianenko opened the pull request.

MishaDemianenko committed:
Migrate license, collections and common modules test to junit 5

andreikoval commented on community/collections/src/test/java/org/neo4j/collection/PrimitiveLongArrayQueueTest.java:
```diff
@@ -204,10 +205,10 @@ public void addAllElementsFromOtherQueue()
+    void doNotAllowCreationOfQueueWithRandomCapacity()
     {
-        new PrimitiveLongArrayQueue( 7 );
+        assertThrows( AssertionError.class, () -> new PrimitiveLongArrayQueue( 7 ) );
```
eliminate assert statement in ctor and replace it with IAE

andreikoval commented on community/collections/src/test/java/org/neo4j/collection/pool/LinkedQueuePoolTest.java:
```diff
@@ -339,11 +338,11 @@ public void release()
-        public AtomicInteger targetSize = new AtomicInteger( -1 );
+        AtomicInteger currentPeakSize = new AtomicInteger( -1 );
+        AtomicInteger targetSize = new AtomicInteger( -1 );
         public AtomicInteger created = new AtomicInteger( 0 );
```
do these still need to be public?

andreikoval commented on community/collections/src/test/java/org/neo4j/helpers/collection/TestCommonIterators.java:
```diff
@@ -55,175 +56,175 @@ public void testNoDuplicatesFilteringIterator()
         try
         {
-            Iterators.first( Arrays.asList().iterator() );
+            Iterators.first( asList().iterator() );
```
for consistency pls use `assertThrows` here and in other places

andreikoval commented on community/common/src/test/java/org/neo4j/ports/allocation/PortRepositoryIT.java:
```diff
@@ -69,8 +71,8 @@ public void shouldCoordinateUsingFileSystem() throws Exception
     @Test
-    @Ignore
-    public void shouldNotInterfereWithOtherRepositories() throws Exception
+    @Disabled
```
maybe delete it?

MishaDemianenko committed:
PR reviews:
use assertThrows in more places, remove long time ignored tests,
use IAE instead of assert keyword

MishaDemianenko commented:
@andreikoval here we go. All comments should be fixed now.

andreikoval commented on community/collections/src/main/java/org/neo4j/collection/PrimitiveLongArrayQueue.java:
```diff
@@ -45,7 +45,10 @@ public PrimitiveLongArrayQueue()
-        assert (capacity != 0) && ((capacity & (capacity - 1)) == 0) : "Capacity should be power of 2.";
+        if ( capacity == 0 || (capacity & (capacity - 1)) != 0 )
+        {
+            throw new IllegalArgumentException( "Capacity should be power of 2." );
```
maybe add actual value to the error message?

MishaDemianenko replied:
yes updated

MishaDemianenko committed:
Update exception message to include incorrect requested capacity.
